rustvmm_gen/lib_kernel_source.py

```diff
@@ -0,0 +1,136 @@
+# Copyright 2025 © Institute of Software, CAS. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+import os
+import re
+import tarfile
+import requests
+import subprocess
+import shutil
+from urllib.parse import urlparse
+
+KERNEL_ORG_CDN = "https://cdn.kernel.org/pub/linux/kernel"
+
+
+def check_kernel_version(version):
+    """
+    Validate if the input kernel version exists in remote. Supports both X.Y
+    (namely X.Y.0 and .0 should be omitted) and X.Y.Z formats
+    """
+    # Validate version format
+    if not re.match(r"^\d+\.\d+(\.\d+)?$", version):
+        return (False, "Invalid version format. Use X.Y or X.Y.Z")
```

Comment:
Why this tuple of string and bool instead of just raising exceptions for errors, and returning void?